pkg/util/wildcard/match_test.go

```diff
@@ -0,0 +1,542 @@
+		},
+		// Test case - 31.
+		{
+			pattern: "my-bucket/abc????",
+			text:    "my-bucket/abcdefg",
+			matched: true,
+		},
+		// Test case 32-34.
+		// test case with no '*'.
+		{
+			pattern: "my-bucket/abc?",
+			text:    "my-bucket/abc",
+			matched: false,
+		},
+		{
+			pattern: "my-bucket/abc?",
+			text:    "my-bucket/abcd",
+			matched: true,
+		},
+		{
+			pattern: "my-bucket/abc?",
+			text:    "my-bucket/abcde",
+			matched: false,
+		},
+		// Test case 35.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mnop",
+			matched: false,
+		},
+		// Test case 36.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mnopqrst/mnopqr",
+			matched: true,
+		},
+		// Test case 37.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mnopqrst/mnopqrs",
+			matched: true,
+		},
+		// Test case 38.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mnop",
+			matched: false,
+		},
+		// Test case 39.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mnopq",
+			matched: true,
+		},
+		// Test case 40.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mnopqr",
+			matched: true,
+		},
+		// Test case 41.
+		{
+			pattern: "my-bucket/mnop*?and",
+			text:    "my-bucket/mnopqand",
+			matched: true,
+		},
+		// Test case 42.
+		{
+			pattern: "my-bucket/mnop*?and",
+			text:    "my-bucket/mnopand",
+			matched: false,
+		},
+		// Test case 43.
+		{
+			pattern: "my-bucket/mnop*?and",
+			text:    "my-bucket/mnopqand",
+			matched: true,
+		},
+		// Test case 44.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mn",
+			matched: false,
+		},
+		// Test case 45.
+		{
+			pattern: "my-bucket/mnop*?",
+			text:    "my-bucket/mnopqrst/mnopqrs",
+			matched: true,
+		},
+		// Test case 46.
+		{
+			pattern: "my-bucket/mnop*??",
+			text:    "my-bucket/mnopqrst",
+			matched: true,
+		},
+		// Test case 47.
+		{
+			pattern: "my-bucket/mnop*qrst",
+			text:    "my-bucket/mnopabcdegqrst",
+			matched: true,
+		},
+		// Test case 48.
+		{
+			pattern: "my-bucket/mnop*?and",
+			text:    "my-bucket/mnopqand",
+			matched: true,
+		},
+		// Test case 49.
+		{
+			pattern: "my-bucket/mnop*?and",
+			text:    "my-bucket/mnopand",
+			matched: false,
+		},
+		// Test case 50.
+		{
+			pattern: "my-bucket/mnop*?and?",
+			text:    "my-bucket/mnopqanda",
+			matched: true,
+		},
+		// Test case 51.
+		{
+			pattern: "my-bucket/mnop*?and",
+			text:    "my-bucket/mnopqanda",
+			matched: false,
+		},
+		// Test case 52.
+
+		{
+			pattern: "my-?-bucket/abc*",
+			text:    "my-bucket/mnopqanda",
+			matched: false,
+		},
+	}
+	// Iterating over the test cases, call the function under test and asert the output.
+	for i, testCase := range testCases {
+		actualResult := wildcard.Match(testCase.pattern, testCase.text)
+		if testCase.matched != actualResult {
+			t.Errorf("Test %d: Expected the result to be `%v`, but instead found it to be `%v`", i+1, testCase.matched, actualResult)
+		}
```

Comment:
```suggestion
		t.Run(fmt.Sprintf("%d", i+1), func(t *testing.T) {
			actualResult := wildcard.Match(testCase.pattern, testCase.text)
			require.Equal(t, actualResult, testCase.matched)
		})
```
It's a personal preference, but I'd prefer something like this.